serialization/python/json/serialization/json_parse_node.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+from datetime import date, datetime, time, timedelta
+from enum import Enum
+from io import BytesIO
+from typing import Any, Callable, Generic, List, Optional, Type, TypeVar
+from uuid import UUID
+
+from dateutil import parser
+from kiota.abstractions.serialization import Parsable, ParseNode
+
+T = TypeVar("T")
+
+U = TypeVar("U", bound=Parsable)
+
+K = TypeVar("K", bound=Enum)
+
+
+class JsonParseNode(ParseNode, Generic[T, U, K]):
+
+    on_before_assign_field_values: Optional[Callable[[Parsable], None]]
+    on_after_assign_field_values: Optional[Callable[[Parsable], None]]
+
+    def __init__(self, node: Any) -> None:
```

Comment:
does the json lib you're using expose a type for the json element?

Reply:
Nope. No default type for JSON. Unless you configure a custom type like so `JSON = Union[Dict[str, Any], List[Any], int, str, float, bool, Type[None]]`